qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1105,20 +1144,27 @@ def to_dataframe(self):
         """
         with qdb.sql_connection.TRN:
             # Retrieve all the information from the database
-            sql = "SELECT * FROM qiita.{0}".format(self._table_name(self._id))
+            cols = self.categories()
+            sql = """SELECT sample_id, sample_values
+                     FROM qiita.{0}
+                     WHERE sample_id != '{1}'""".format(
+                        self._table_name(self._id), QIITA_COLUMN_NAME)
             qdb.sql_connection.TRN.add(sql)
-            meta = qdb.sql_connection.TRN.execute_fetchindex()
 
-            # When we create the dataframe, we are providing a matrix (list of
-            # lists) with all the data, so we need to make sure that all the
-            # rows contain the columns in the same order
-            cols = sorted(meta[0].keys())
-            meta_matrix = [[r[c] for c in cols] for r in meta]
-            df = pd.DataFrame(meta_matrix, columns=cols, dtype=str)
+            # we could have used dict() and then use pandas from_dict but
+            # that takes forever so it's much faster to simply parse here and
+            # load via pandas.DataFrame
+            data = []
+            for r in qdb.sql_connection.TRN.execute_fetchindex():
+                v = [r[0]]
+                v.extend([r[1][c] if c in r[1] else None for c in cols])
```

Comment:
I have no idea what this is doing

Reply:
Isn't the `c in r[1]` a list lookup?

it would really help to break this up and not use a dict comprehension, or to break up the magic numbers and assign variable names